tests/test_main.py

```diff
@@ -168,18 +168,20 @@ def test_main_session_from_nox_env_var(monkeypatch, env, sessions):
 
 def test_main_positional_args(monkeypatch):
     monkeypatch.setattr(sys, "argv", [sys.executable, "1", "2", "3"])
-    with mock.patch("nox.workflow.execute") as execute:
-        execute.return_value = 0
 
-        # Call the main function.
-        with mock.patch.object(sys, "exit") as exit:
-            nox.__main__.main()
-            exit.assert_called_once_with(0)
-        assert execute.called
+    with pytest.raises(
+        AttributeError, match="object has no attribute 'color'"
```

Comment:
I don't understand this test anymore. What's happening here?

Reply:
The `_exit` function was a no-op, so instead of exiting, nox continued executing `main()` until it hit the next exception which was an attribute error. Since the arguments were not parsed, `args` didn't have the `color` attribute. That is pretty fragile and unintuitive, so I updated with a mocked exit that raises an exception so the unit test can catch it.